internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -675,6 +676,37 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
 				},
 			},
 
+			"key_vault_kms": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				ForceNew: false,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"enabled": {
```

Comment:
We don't expose this field in the provider since the [presence of the block implies that the feature should be enabled and the absence of it means disabled](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/schema-design-considerations.md#features-that-are-toggled-by-the-property-enabled) so this will need to be removed.